internal/grpc/services/userprovider/userprovider.go

```diff
@@ -99,19 +100,36 @@ func (s *service) Register(ss *grpc.Server) {
 }
 
 func (s *service) GetUser(ctx context.Context, req *userpb.GetUserRequest) (*userpb.GetUserResponse, error) {
-	user, err := s.usermgr.GetUser(ctx, req.UserId)
-	if err != nil {
-		// TODO(labkode): check for not found.
-		err = errors.Wrap(err, "userprovidersvc: error getting user")
-		res := &userpb.GetUserResponse{
-			Status: status.NewInternal(ctx, err, "error authenticating user"),
+
+	// Check if we need to retrieve user from UID or not
+	uid, err := extractUID(req.Opaque)
```

Comment:
To avoid this, we should create a new method on the CS3APIS to return a User by using other claims, for example:

```
GetUserByClaim
```